### ISSUE TITLE
Launching from WSL failing

### ISSUE THREAD

**Issue by david-engelmann** [microsoft/vscode-remote-release#8137]
- VSCode Version: 

`Version: 1.76.0 (user setup)`
`Commit: 92da9481c0904c6adfe372c12da3b7748d74bdcb`
`Date: 2023-03-01T10:22:44.506Z`
`Electron: 19.1.11`
`Chromium: 102.0.5005.196`
`Node.js: 16.14.2`
`V8: 10.2.154.26-electron.0`
`OS: Windows_NT x64 10.0.22000`
`Sandboxed: No`

- Local OS Version: Windows 11 Pro
- Local chip architecture: 64-bit operating system, x64-based processor
- Reproduces in: Remote - Containers
- Name of Dev Container Definition with Issue: Dev Container Config

Steps to Reproduce:

1. Launch a project with a devcontainer.json in VsCode connected to the WSL Server
- Ie. Go to the folder in wsl or Ubuntu and `code .`
2. Reopen in Dev Container from VsCode WSL Server

Error:

`Error: Dev container config (c:\mnt\c\users\........`

For some reason when launching from wsl, its looking for the devcontainer.json in c:\mnt\c

Does this issue occur when you try this locally?: Occurs with Docker Desktop on WSL
Does this issue occur when you try this locally and all extensions are disabled?: Yes

[Related Issue](https://github.com/microsoft/vscode-dev-containers/issues/1722)

**Comment by chrmarti** [comment]
Could you append the container log from when this fails? (`F1` > `Dev Containers: Show Container Log`)

**Comment by david-engelmann** [comment]
[remoteContainers-2023-03-07T21-40-18.973Z.log](https://github.com/microsoft/vscode-remote-release/files/10914158/remoteContainers-2023-03-07T21-40-18.973Z.log)

**Comment by chrmarti** [comment]
Do you have any of these two settings set?
<img width="931" alt="image" src="https://user-images.githubusercontent.com/9205389/223676547-214bc5fc-5195-4f70-b34e-197d24ee9357.png">

**Comment by david-engelmann** [comment]
![image](https://user-images.githubusercontent.com/81196676/223739269-4e61dfec-6660-4d9d-a41d-59ef979744d3.png)

I have them but they are not selected

**Comment by david-engelmann** [comment]
@chrmarti Any progress here?

**Comment by chrmarti** [comment]
No, we're using the Node version that is part of Electron and pass in NODE_PATH, so node-pty is found.

**Comment by stuartleeks** [comment]
It looks as though you have the code on your Windows drive? `c:\mnt\c\users\David Engelmann\VsCodeProjects\external_base\`

I'm not sure why that would cause the issue, but in general if you are able to put the code in a folder in the WSL filesystem you will get better performance and compatibility (e.g. file permissions). It might also be interesting to test to see if you get the same issue in that setup?

**Comment by david-engelmann** [comment]
@stuartleeks This is a great point, and seems to change the results of launching from wsl and from an active devcontainer. Fortunately my references were all relative, however, I did have to make some adjustments to the ssh logic in my Dockerfile.

```
RUN chown $(whoami) /root/.ssh/config && \
    chmod 644 /root/.ssh/config
```
This means the problem is isolated to launching from the \mnt\c directory. This was extremely useful as a test but as far as only using a folder in the WSL Filesystem, its not that practical. Some users are not that familiar with WSL, and work predominantly within windows, as well as the results are usually needed within the windows filesystem. Additionally, volumes and mounted files might exist in the windows filesystem.

**Comment by david-engelmann** [comment]
@chrmarti Any progress here?

**Comment by chrmarti** [comment]
Found it, this seems to be a recent regression. Preparing a fix for the next pre-release version.

**Comment by david-engelmann** [comment]
@chrmarti Amazing!!
